src/torchmetrics/wrappers/multitask.py

```diff
@@ -0,0 +1,110 @@
+from typing import Any, Dict, Optional, Sequence, Union
+
+from torch import Tensor
+
+from torchmetrics import Metric, MetricCollection
+from torchmetrics.utilities.imports import _MATPLOTLIB_AVAILABLE
+from torchmetrics.utilities.plot import _AX_TYPE, _PLOT_OUT_TYPE
+
+if not _MATPLOTLIB_AVAILABLE:
+    __doctest_skip__ = ["MultitaskWrapper.plot"]
+
+
+class MultitaskWrapper(Metric):
+    """Wrapper class for computing different metrics on different tasks in the context of multitask learning.
+
+    It computes a different metric on each pred and target given to its `update` method.
+
+    Args:
+        task_metrics:
+            Dictionary associating each task to a Metric or a MetricCollection. The keys of the dictionary represent the
+            names of the tasks, and the values represent the metrics to use for each task.
+
+    Example:
+         >>> import torch
+         >>> from torchmetrics.wrappers import MultitaskWrapper
+         >>> from torchmetrics.regression import MeanSquaredError
+         >>> from torchmetrics.classification import BinaryAccuracy
+         >>> classification_target = torch.tensor([0, 1, 0])
+         >>> regression_target = torch.tensor([2.5, 5.0, 4.0])
+         >>> targets = {"Classification": classification_target, "Regression": regression_target}
+         >>> classification_preds = torch.tensor([0, 0, 1])
+         >>> regression_preds = torch.tensor([3.0, 5.0, 2.5])
+         >>> preds = {"Classification": classification_preds, "Regression": regression_preds}
+         >>> metrics = MultitaskWrapper({
+         ...     "Classification": BinaryAccuracy(),
+         ...     "Regression": MeanSquaredError()
+         ... })
+         >>> metrics.update(preds, targets)
+         >>> metrics.compute()
+         {'Classification': tensor(0.3333), 'Regression': tensor(0.8333)}
+    """
+
+    is_differentiable = False
+
+    def __init__(
+        self,
+        task_metrics: Dict[str, Union[Metric, MetricCollection]],
+    ) -> None:
+        self._check_task_metrics_type(task_metrics)
+        super().__init__()
+        self.task_metrics = task_metrics
```

Comment:
```suggestion
        self.task_metrics = nn.ModuleDict(task_metrics)
```
I am 99% sure we need to do a wrapping here, because if not the other metrics will not be considered "children" of the wrapper and therefore calling `.cuda()`, `.to()` etc. method will not move the states of the sub-modules to the correct device

Reply:
I just quickly tested it and you're right, I'll fix that!